src/CMakeLists.txt

```diff
@@ -18,10 +18,10 @@ target_include_directories(cps PUBLIC "${CMAKE_CURRENT_BINARY_DIR}")
 
 # Dependencies
 find_package(tl-expected 1.0 REQUIRED)
-target_link_libraries(cps PRIVATE tl::expected)
+target_link_libraries(cps PUBLIC tl::expected)
 
 find_package(fmt 8 REQUIRED)
-target_link_libraries(cps PRIVATE fmt::fmt)
+target_link_libraries(cps PUBLIC fmt::fmt)
```

Comment:
fmt shouldn't be leaking into the public scope, the only header it's in is utils.hpp... and that could be moved into a source file. I'll send a patch to clean that header up, it's kinda messy.

Otherwise this looks fine to me.

Reply:
Ah I didn't see you made that other PR